<h2>Why is this an issue?</h2>
<p>The purpose of an abstract class is to provide some heritable behaviors while also defining methods which must be implemented by sub-classes.</p>
<p>A <code>class</code> with only <code>abstract</code> methods and no inheritable behavior should be converted to an <code>interface</code>.</p>
<h3>Noncompliant code example</h3>
<pre>
public abstract class Animal // Noncompliant; should be an interface
{
  abstract void Move();
  abstract void Feed();
}
</pre>
<h3>Compliant solution</h3>
<pre>
public interface Animal
{
  void Move();
  void Feed();
}

public abstract class Lamp
{
  private bool switchLamp = false;

  public abstract void Glow();

  public void FlipSwitch()
  {
    switchLamp = !switchLamp;
    if (switchLamp)
    {
      Glow();
    }
  }
}
</pre>

